interfaces/builtin/sd_control.go

```diff
@@ -0,0 +1,90 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/udev"
+)
+
+const sdControlSummary = `allows controlling SD cards on certain boards`
+
+const sdBaseDeclarationSlots = `
+  sd-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - app
+    deny-auto-connection: true
+`
+
+const dualSDSDControlConnectedPlugApparmor = `
+# Description: can manage and control the SD cards using the DualSD driver.
+
+# The main DualSD device node is used to control certain aspects of SD cards on
+# the system.
+/dev/DualSD rw,
```

Comment:
Could this be rolled into the block devices maybe? At least we should mention what driver we want to support here, eg. vendor/name or the device. I was unable to find anything useful googling for /dev/DualSD.

Reply:
well the /dev/DualSD device is not a block device, it is sort of block-devices related, but since it is so specific to this board/hardware we felt it best to have a specific interface for it like this. And yeah I couldn't find anything else out either :-/